lib/pure/json.nim

```diff
@@ -531,6 +531,26 @@ proc `[]`*(node: JsonNode, index: BackwardsIndex): JsonNode {.inline, since: (1,
 
   `[]`(node, node.len - int(index))
 
+proc `[]`*[U, V](a: JsonNode, x: HSlice[U, V]): JsonNode =
+  ## Slice operation for JArray.
+  ## Returns the inclusive range `[a[x.a], a[x.b]]`:
+  ##
+  runnableExamples:
+    let arr = %[0,1,2,3,4,5]
+    doAssert arr[2..4] == %[2,3,4]
+    doAssert arr[2..^2] == %[2,3,4]
+
+  if a.kind != JArray:
```

Comment:
The other operations simply use `assert(node.kind == JArray)`

Reply:
@Araq 
I have fixed.